joblib/_memmapping_reducer.py

```diff
@@ -251,8 +251,8 @@ def __call__(self, a):
 
             # Find a unique, concurrent safe filename for writing the
             # content of this array only once.
-            basename = "{}-{}-{}.pkl".format(
-                os.getpid(), id(threading.current_thread()), hash(a))
+            basename = "{}-{}-{}-{}.pkl".format(
+                os.getpid(), id(threading.current_thread()), hash(a), id(a))
```

Comment:
I seemed to remember that we wanted to use `id(a)` only in the case where the memmap was in write mode.

Reply:
I'm not sure I follow. I thought this was the change we'd discussed before. Are you suggesting that we check the `self._mmap_mode` and include/not include the id based on that? Is there an advantage to not always using `id(a)` here?